packages/synapse-react-client/src/components/WikiMarkdownEditor/WikiMarkdownEditor.tsx

```diff
@@ -0,0 +1,256 @@
+}
+
+export const WikiMarkdownEditor: React.FunctionComponent<
+  WikiMarkdownEditorProps
+> = (props: WikiMarkdownEditorProps) => {
+  // TODO: SWC-6774 - get showDeleteButton from props
+  const showDeleteButton = false
+  const {
+    ownerObjectType,
+    ownerObjectId,
+    wikiPageId: initialWikiPageId,
+    onCancel = noop,
+    onSave = noop,
+  } = props
+
+  const [wikiPageId, setWikiPageId] = useState<string | undefined>(
+    initialWikiPageId,
+  )
+  const [open, setOpen] = useState<boolean>(true)
+  const [title, setTitle] = useState<string>('')
+  const [markdown, setMarkdown] = useState<string>('')
+  const [showConfirmCancelDialog, setShowConfirmDialog] =
+    useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const { data: rootWikiPageKey, error: errorLoadingRootWikiPageKey } =
+    useGetRootWikiPageKey(ownerObjectType, ownerObjectId, {
+      enabled: wikiPageId === undefined,
+    })
+
+  const { mutate: createWikiPage } = useCreateWikiPage({
+    onSuccess: wikiPage => {
+      setError(null)
+      setWikiPageId(wikiPage.id)
+    },
+    onError: error => {
+      setError(error.reason)
+    },
+  })
+
+  useEffect(() => {
+    setError(null)
+
+    // an existing root WikiPageKey was found
+    if (rootWikiPageKey) {
+      setWikiPageId(rootWikiPageKey.wikiPageId)
+    }
+
+    // root WikiPageKey was not found (so we have create permission),
+    // then create a root WikiPage for this ownerObject
+    if (rootWikiPageKey === null) {
+      const rootWikiPage: CreateWikiPageInput['wikiPage'] = {
+        parentWikiId: undefined,
+        title: '',
+        markdown: '',
+        attachmentFileHandleIds: [],
+      }
+      const input: CreateWikiPageInput = {
+        ownerObjectId,
+        ownerObjectType,
+        wikiPage: rootWikiPage,
+      }
+      createWikiPage(input)
+    }
+
+    // any other error finding root WikiPageKey
+    if (errorLoadingRootWikiPageKey) {
+      setError(ERROR_LOADING_WIKI_FAILED + errorLoadingRootWikiPageKey.reason)
+    }
+  }, [
+    rootWikiPageKey,
+    ownerObjectId,
+    ownerObjectType,
+    errorLoadingRootWikiPageKey,
+    createWikiPage,
+  ])
+
+  const wikiPageKey = useMemo(() => {
+    const wikiPageKey: WikiPageKey = {
+      ownerObjectType,
+      ownerObjectId,
+      wikiPageId: wikiPageId || '',
+    }
+    return wikiPageKey
+  }, [wikiPageId, ownerObjectId, ownerObjectType])
+
+  const { data: wikiPage, error: errorLoadingWikiPage } = useGetWikiPage(
+    wikiPageKey,
+    {
+      enabled: wikiPageId !== undefined,
+      // Set staleTime to infinity to prevent re-fetching while editing
+      staleTime: Infinity,
+    },
+  )
+
+  useEffect(() => {
+    setError(null)
+
+    if (wikiPage) {
+      setTitle(wikiPage.title)
+      setMarkdown(wikiPage.markdown)
+    }
+    if (errorLoadingWikiPage) {
+      setError(ERROR_LOADING_WIKI_FAILED + errorLoadingWikiPage.reason)
+    }
+  }, [errorLoadingWikiPage, wikiPage])
```

Comment:
Ideally if you can derive the error instead of using a state variable as described above, this would simplify to:
```suggestion
  useEffect(() => {
    if (wikiPage) {
      setTitle(wikiPage.title)
      setMarkdown(wikiPage.markdown)
    }
  }, [wikiPage])
```